core/src/main/scala/filodb.core/metadata/Dataset.scala

```diff
@@ -204,19 +212,17 @@ object DatasetOptions {
  * Contains many helper functions especially pertaining to Dataset creation and validation.
  */
 object Dataset {
-  /**
-   * Re-creates a Dataset from the output of `asCompactString`
-   */
-  def fromCompactString(compactStr: String): Dataset = {
-    val Array(database, name, partColStr, dataColStr, rowKeyIndices, downsamplersStr, optStr) =
-      compactStr.split('\u0001')
-    val partitionColumns = partColStr.split(':').toSeq.map(raw => DataColumn.fromString(raw))
-    val dataColumns = dataColStr.split(':').toSeq.map(raw => DataColumn.fromString(raw))
-    val rowKeyIDs = rowKeyIndices.split(':').toSeq.map(_.toInt)
-    val downsamplers = downsamplersStr.split(':').filterNot(_.isEmpty).toSeq.map(ChunkDownsampler.downsampler(_))
-    val databaseOption = if (database == "") None else Some(database)
-    val options = DatasetOptions.fromString(optStr)
-    Dataset(name, partitionColumns, dataColumns, rowKeyIDs, downsamplers, databaseOption, options)
+  def fromConfig(config: Config): Dataset = {
```

Comment:
Note: Config is much more expensive to serialize than the String.  Though we probably won't be doing this that often is my guess.

Reply:
This is only executed at boot time when we load datasets from config.